compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -2224,6 +2224,9 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             }
         };
 
+        // Regions are not considered during selection.
+        let erased_self_ty = tcx.fold_regions(self_ty, |_, _| tcx.lifetimes.re_erased);
```

Comment:
I'm going back and forth whether we should use `replace_escaping_bound_vars_uncached` here instead... 

If you don't want to use that, can you at least leave a FIXME saying that this only *needs* to deal w escaping bound vars, and it currently doesn't handle ty/ct vars correctly?